include/ck/tensor_operation/gpu/grid/gridwise_batched_gemm_gemm_xdl_cshuffle_v1.hpp

```diff
@@ -216,6 +217,13 @@ struct GridwiseBatchedGemmGemm_Xdl_CShuffle
             return false;
         }
 
+        // K is rounded to nearest multiples of K1 during tensor transformation so instead get KRaw
+        const auto KRaw = lengths_m_n_k_o[2];
+        if(!(KRaw % AK1 == 0 && KRaw % BK1 == 0))
```

Comment:
should check the length with vector length, but AK1 and BK1 is not always the same as vector length

Reply:
Is this what you refer to? Implemented in device level
https://github.com/ROCmSoftwarePlatform/composable_kernel/blob/develop/include/ck/tensor_operation/gpu/device/device_batched_gemm_gemm_xdl_cshuffle.hpp#L600-L616

Reply:
Yes, so there is already this check. Great!

Why not put these 2 checks in the same place? Either in gridwise GEMM or DeviceOp?

Reply:
Ideally I want both checks in gridwise op. But the A/B/B1/C layout is lost in gridwise op (embedded in tensor descriptor). And inferring layout from tensor descriptor should work but isn't immediately obvious to me how. Suggestions?

Reply:
> Actually, why we need to check KRaw % AK1 == 0 && KRaw % BK1 == 0?

@asroy So I think about this again - this check is wrong and actually reject valid cases. Checking with ScalarPerVector is sufficient. I can remove it in next PRs. Although feel free to modify it while you are refactoring gridwise_batched_gemm_gemm

Reply:
removed in PR #377